### PR TITLE
Initial removeWhere.md for dart

### PR DESCRIPTION
**Pull request by lordgabriel98**
### Description
Added documentation for .removeWhere() method for lists in Dart programming language. Also created the directory for list under Dart/concepts and other folders that were mentioned in the issue.

### Issue Solved

 Dart List .removeWhere() #4315

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by avdhoottt** [comment]
Hey @lordgabriel98, the entry is not following the term entry guidelines. Please go through the [term-entry-template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md) and then improvise your content to meet the content guidelines.

**Comment by lordgabriel98** [comment]
@avdhoottt apologies for the delay. 

Updated removeWhere.md to better suit term-entry-template.

**Comment by lordgabriel98** [comment]
@avdhoottt please review this merge request asap. Thank you.

File: content/dart/concepts/list/terms/removeWhere/removeWhere.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  - 'Lists'
+  - 'Methods'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+The **.removeWhere()** method removes elements from a list, based on a given condition that is provided as an argument.
+
+## Syntax
+```pseudo
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/removeWhere/removeWhere.md:17]: Give one line space before and after a heading/code block/meta data.

File: content/dart/concepts/list/terms/removeWhere/removeWhere.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  - 'Computer Science'
+  - 'Web Development'
+Tags:
+  - 'Lists'
+  - 'Methods'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+The **.removeWhere()** method removes elements from a list, based on a given condition that is provided as an argument.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/removeWhere/removeWhere.md:14]:
```suggestion

The **.removeWhere()** method removes elements from a list, based on a given condition that is provided as an argument.
```

File: content/dart/concepts/list/terms/removeWhere/removeWhere.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## Syntax
+```pseudo
+void removeWhere(bool Function(E element) test)
+```
+
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/removeWhere/removeWhere.md:20]: Describe the parameter function `test` below the syntax block.

File: content/dart/concepts/list/terms/removeWhere/removeWhere.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+The '.removeWhere()' method takes in a `bool` function as an argument, where the function defines the condition. Values in the list are removed if the condition evaluates to 'true'.
+## Example
+
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/removeWhere/removeWhere.md:23]: Add the description of the example in one line below the `Example` heading.

File: content/dart/concepts/list/terms/removeWhere/removeWhere.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+```dart
+void main(){
+
+List<int> numbers = [1,2,3,4,5,6,7,8,9,10];
+
+numbers.removeWhere((int number) => number % 2 ==0);
+
+print(numbers); // Output: [2,4,6,8,10]
+}
+
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/removeWhere/removeWhere.md:33]: Close the code block with closing ticks and add the output inside a ` ```shell `.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey @lordgabriel98, sorry for the delay. Please make the suggested changes.

**Comment by lordgabriel98** [comment]
@avdhoottt thank for the review. Please review updates that I have made.

File: content/dart/concepts/list/terms/removeWhere/removeWhere.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+The **.removeWhere()** method removes elements from a list, based on a given condition that is provided as an argument.
+
+## Syntax
+
+```pseudo
+void removeWhere(bool Function(E element) test)
+```
+
+The .removeWhere() method takes in a `bool` function as an argument, referred to as the `test` parameter, which defines the condition for removal. Values in the list are removed if they satisfy this condition, meaning if the `test` function evaluates to `true` for those values.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/removeWhere/removeWhere.md:23]:
```suggestion
- `test`: This function defines the condition for removal. Values in the list are removed if they satisfy this condition, meaning if the `test` function evaluates to `true` for those values.
```

**Comment by lordgabriel98** [comment]
@avdhoottt have made final changes to removeWhere.md.

File: content/dart/concepts/list/terms/removeWhere/removeWhere.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+The **`.removeWhere()`** method removes elements from a list, based on a given condition that is provided as an argument.
+
+## Syntax
+
+```pseudo
+void removeWhere(bool Function(E element) test)
+```
+
+- `test`: This function defines the condition for removal. Values in the list are removed if they satisfy this condition, meaning if the `test` function evaluates to `true` for those values
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/removeWhere/removeWhere.md:23]:
```suggestion
- `test`: This function defines the condition for removal. Values in the list are removed if they satisfy this condition, meaning if the `test` function evaluates to `true` for those values.
```

File: content/dart/concepts/list/terms/removeWhere/removeWhere.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+List<int> numbers = [1,2,3,4,5,6,7,8,9,10];
+
+numbers.removeWhere((int number) => number % 2 ==0);
+
+print(numbers);
+}
+```
+
+The above code will produce the following output:
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/removeWhere/removeWhere.md:40]:
```suggestion
The above code will return the following output:
```

File: content/dart/concepts/list/terms/removeWhere/removeWhere.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+This example demonstrates using `.removeWhere()` to remove even numbers from a list, resulting in a list of odd numbers.
+
+```dart
+void main() {
+  List<int> numbers = [1,2,3,4,5,6,7,8,9,10];
+  numbers.removeWhere((int number) => number % 2 ==0);
+  print(numbers);
+}
+
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/removeWhere/removeWhere.md:35]:
```suggestion
```

**Review by avdhoottt — APPROVED** [review]
Looks good for the second review!! Thanks.

File: content/dart/concepts/list/terms/removeWhere/removeWhere.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+The **`.removeWhere()`** method removes elements from a list, based on a given condition that is provided as an argument.
+
+## Syntax
+
+```pseudo
+void removeWhere(bool Function(E element) test)
```
Comment (ishg-153) [inline content/dart/concepts/list/terms/removeWhere/removeWhere.md:20]:
```suggestion
void removeWhere(bool Function(E element))
```

File: content/dart/concepts/list/terms/removeWhere/removeWhere.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+- `test`: This function defines the condition for removal. Values in the list are removed if they satisfy this condition, meaning if the `test` function evaluates to `true` for those values.
```
Comment (ishg-153) [inline content/dart/concepts/list/terms/removeWhere/removeWhere.md:23]:
```suggestion
- `Function(E element)`: It defines the condition for removal. If the function evaluates to `true` for a value in the list, then the value is removed from the list.
```

**Comment by lordgabriel98** [comment]
@ishg-153 I have made modifications as suggested. Thanks

**Review by ishg-153 — APPROVED** [review]
@lordgabriel98 Looks good to merge!

**Comment by lordgabriel98** [comment]
@ishg-153 roger that. Merging removewhere branch into main

**Pull request merged**